pkg/route/validation/validation.go

```diff
@@ -342,6 +370,41 @@ func validateWildcardPolicy(host string, policy routev1.WildcardPolicyType, fldP
 	return nil
 }
 
+var (
+	notAllowedHTTPHeaderResponse    = []string{"strict-transport-security"}
+	notAllowedHTTPHeaderResponseSet = sets.NewString(notAllowedHTTPHeaderResponse...)
+)
+
+func validateHeaders(path *field.Path, headerValues []routev1.RouteHTTPHeader) *field.Error {
+	for key, value := range headerValues {
+		if value.Action.Set != nil && len(value.Action.Set.Value) != 0 && len(value.Action.Set.Value) > 4096 {
+			return field.Invalid(path.Index(key).Child("action.set.value"), value.Name, fmt.Sprintf("value exceeds the maximum length which is 2048"))
+		}
+		if value.Action.Set != nil && len(value.Action.Set.Value) == 0 {
+			return field.Invalid(path.Index(key).Child("action.set.value"), value.Name, fmt.Sprintf("minimum length of value shall be one."))
+		}
+		if path.String() == "spec.httpHeaders.actions.response" && value.Action.Set != nil && len(value.Action.Set.Value) != 0 {
+			r := regexp.MustCompile("^(?:%(?:%|(?:\\{[-+]?[QXE](,[-+]?[QXE])*\\})?\\[(?:res\\.hdr\\([0-9A-Za-z-]+\\)|ssl_c_der)(?:,(?:lower|base64))*\\])|[^%[:cntrl:]])*$")
+
+			if !r.MatchString(value.Action.Set.Value) {
+				return field.Invalid(path.Index(key).Child("action.set.value"), value.Name, "Either header value provided is not in correct format or the converter specified is not allowed. The dynamic header value will be interpreted as an HAProxy format string as defined in http://cbonte.github.io/haproxy-dconv/2.6/configuration.html#8.2.6 and may use HAProxy's %[] syntax and otherwise must be a valid HTTP header value as defined in https://datatracker.ietf.org/doc/html/rfc7230#section-3.2 Sample fetchers allowed are req.hdr, ssl_c_der. Converters allowed are lower, base64.")
```

Comment:
Also, https://datatracker.ietf.org/doc/html/rfc7230#section-3.2.2 says the following, which I don't think we want to allow.  You probably should not include a reference to that rfc7230 unless we follow all of the standard.  As I mentioned above, I'd like to see some kind of list of what can and cannot be used, rather than a pointer to a rfc from which we took some requirements, but not all requirements.

> A recipient MAY combine multiple header fields with the same field
>    name into one "field-name: field-value" pair, without changing the
>    semantics of the message, by appending each subsequent field value to
>    the combined field value in order, separated by a comma.  The order
>    in which header fields with the same field name are received is
>    therefore significant to the interpretation of the combined field
>    value; a proxy MUST NOT change the order of these field values when
>    forwarding a message.

Reply:
updated it.